Topics/Development_Process/React_Testing_Library.md

```diff
@@ -0,0 +1,35 @@
+## Writing tests with React Testing Library
+
+When writing test suites for React components, it is a good idea to write tests that do not depend on the implementation details of the components (e.g. props, attributes, state, etc.) but rather on what the components are and how they behave. For example, instead of querying an element by its ID attribute, use an accessibility role (e.g. button, list, listitem) or accessibility label. This makes tests less fragile and prone to breaking when small changes in the components are made.
+
+Additionally, it is a good idea to design tests that mimic the way a user would interact with the app. For example, when writing tests for a Login page/component, to test that login is successful on submission of correct credentials, the test should rougly follow the steps:
```

Comment:
A good addition might be to mention _why_ the test should follow the steps you provided. It may not be obvious for people knew to frontend tests.